webapp/sources/rudder/rudder-core/src/main/resources/hooks.d/policy-generation-finished/60-trigger-node-update

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Hooks parameter are passed by environment variable:
+#
+# - RUDDER_GENERATION_DATETIME       : generation datetime: ISO-8601 YYYY-MM-ddTHH:mm:ss.sssZ date/time that identify that policy generation.
+# - RUDDER_END_GENERATION_DATETIME   : generation end time: ISO-8601 YYYY-MM-ddTHH:mm:ss.sssZ date/time when the generation ended (minus these hooks)
+# - RUDDER_NODE_IDS                  : space separated list of node id updated during the process, or the empty string
+#                                      if no nodes were updated.
+# - RUDDER_NUMBER_NODES_UPDATED      : integer >= 0; number of nodes updated (could be found by counting $RUDDER_NODE_IDS)
+# - RUDDER_ROOT_POLICY_SERVER_UPDATED: 0 if root was updated, anything else if not
+
+
+#
+# This hook start a remote run for updated nodes.
+# This hook uses the remote-run API, so you need to open relevant port, 
+# see documentation for "Trigger remote agent run" in
+# https://docs.rudder.io/reference/current/installation/requirements.html#configure-the-network
+# 
+# This feature can generate a more run than expected and some load on the network. In big
+# Rudder installation (thousands of nodes), we advice to measure these metrics and disable
+# the hook if necessary. 
+# 
+
+# Signal to cf-serverd that it shall look-up the new promises
+if [ -n "${RUDDER_NODE_IDS}" ]; then
+  
+  NODES=$(echo "${RUDDER_NODE_IDS}"| sed -r -e "s/[ ]+/,/g")
+  curl -X POST 'http://localhost/rudder/relay-api/remote-run/nodes' -d "nodes=${NODES}" -d 'asynchronous=true' -d 'keep_output=false'
```

Comment:
this will explode if we have too many nodes (limit on command line size) => use a file for post args